client/src/pages/DashBoard.js

```diff
@@ -1,23 +1,65 @@
-import React, { useContext } from 'react';
-
+import React, { useContext, useEffect, useState } from 'react';
 import { useStyles } from '../themes/theme';
 import PastInterviewTable from '../components/PastInterviewTable';
+import UpcomingInterviewTable from '../components/UpcomingInterviewTable'
 import { store } from '../context/store';
-
 import { useHistory } from 'react-router';
 import InterviewDifficultyMenu from './InterviewDifficultyMenu';
-
 import UserInformation from '../components/UserInformation';
-import { createInterview } from '../utils/apiFunctions';
+import { createInterview, getQuestion, getUser, getUserInterviews } from '../utils/apiFunctions';
 import { CustomButton } from '../components/Buttons';
 import { Dialog, DialogTitle } from '@material-ui/core';
 
 const DashBoard = () => {
   const classes = useStyles();
   const history = useHistory();
-  const [open, setOpen] = React.useState(false);
-  const [selectedValue, setSelectedValue] = React.useState('Medium');
+  const [open, setOpen] = useState(false);
+  const [selectedValue, setSelectedValue] = useState('Medium');
   const { state } = useContext(store);
+  const [pageData, setPageData] = useState({
+    pageLoaded: false,
+    interviews: null
+  });
+
+  const fetchInterviews = async (userId) => {
+    const { data: userInterviews } = await getUserInterviews(userId);
+    const interviews = [];
+    for (const interview of userInterviews) {
+      for (const user of interview.users) {
+        if (user.user === userId) {
+          const {
+            _id: userId,
+            firstName,
+            lastName
+          } = (await getUser(user)).data.user;
+
+          const {
+            title: questionTitle,
+            description: questionDescription
+          } = user.question ? (await getQuestion(user.question)).data : {};
+
+          interviews.push({
```

Comment:
Not all of this is being used by the child component, so I don't think we need to store all of it